### PR TITLE
[Rust] numbers - replace about.md file with concept files

### PR DESCRIPTION
**Pull request by ErikSchierboom**
In [this issue](https://github.com/exercism/v3/issues/2293) we're describing an evolution of the specification of this repo.
These changes include, amongst others, replacing the contents of the `after.md` document with individual concept documents. There are two documents per concept:

1. An `about.md` file containing the description of the concept
1. A `links.json` file containing a set of useful links related to the concept. 

This looks as follows in the repo:

```
<track>
├── concepts
│   ├── <concept>
│   │   ├── about.md
│   │   └── links.json
│   └── ...
```

This PR applies this change to this exercise by: 
                
1. Copying the existing `after.md` file's contents to the `about.md` file of each concept listed in the exercise's `concepts` section in the `config.json` file.
1. Extract the links from the `after.md` file and put them into the `links.json` file

Before merging this PR, please check that the contents of the concept documents makes sense.
This is especially true when the exercise unlocks multiple concepts, as then the concept documents should be updated to only contain the information relevant to that concept.

Note: to make reviewing easier, the PR has been split into three separate commits. Please squash this PR upon merging.

### CODE REVIEW

**Review by coriolinus — CHANGES_REQUESTED** [review]
Unacceptable in current form. `integers` and `floating-point numbers` need to be rewritten into topic-specific documentation. `numbers` needs a rewrite to avoid stilted/awkward prose.

**Comment by ErikSchierboom** [comment]
Thanks for the review! It is at this point that the Rust maintainers should take it upon them to improve the text in this document. Good luck!

**Comment by coriolinus** [comment]
Yeah, I know. The Changes Requested marker is as much a guard against accidental merge as anything.

**Comment by ErikSchierboom** [comment]
> Yeah, I know. The Changes Requested marker is as much a guard against accidental merge as anything.

Excellent!

**Comment by coriolinus** [comment]
@exercism/rust as I've now rewritten the text of these concepts, someone else should review that text.

File: languages/rust/concepts/floating-point-numbers/about.md
Comment (petertseng) [inline languages/rust/concepts/floating-point-numbers/about.md:25]: just as there was an example of `expr as Type` above, perhaps useful to show an example here? `f32::from(expr)` ?

File: languages/rust/concepts/integers/about.md
Comment (petertseng) [inline languages/rust/concepts/integers/about.md:47]: same comment about show example of use

File: languages/rust/concepts/numbers/about.md
Comment (petertseng) [inline languages/rust/concepts/numbers/about.md:9]: although the footnote doesn't work in GitHub's view, shall I assume that it will work on the Exercism website and approve anyway? https://github.com/exercism/v3/blob/a5308ec1ca019deb6a40aa35b33630fa769796d5/languages/rust/concepts/numbers/about.md

I see that this footnote syntax is in compliance with https://www.markdownguide.org/extended-syntax/, so I guess I'll hope that the website knows.

**Review by petertseng — APPROVED** [review]

File: languages/rust/concepts/numbers/about.md
Comment (coriolinus) [reply languages/rust/concepts/numbers/about.md:9]: Good point. I don't know whether the exercism markdown renderer supports footnote syntax. @iHiD do you know?

Comment (ErikSchierboom) [reply languages/rust/concepts/numbers/about.md:9]: We're using the [CommonMarker gem](https://www.gjtorikian.com/commonmarker/) to render markdown. As per its docs:

> Ruby wrapper for libcmark-gfm, GitHub's fork of the reference parser for CommonMark. It passes all of the C tests, and is therefore spec-complete. It also includes extensions to the CommonMark spec as documented in the GitHub Flavored Markdown spec, such as support for tables, strikethroughs, and autolinking.

The `:FOOTNOTES` option is currently not enabled as far as I could tell. @iHiD is that correct?

Comment (iHiD) [reply languages/rust/concepts/numbers/about.md:9]: I think we can turn it on. I'll try closer to the time. If it turns out that we can't for whatever reason, then we can fix this "manually" then? (ie, happy to merge)

Comment (coriolinus) [reply languages/rust/concepts/numbers/about.md:9]: Great, thanks!

**Pull request merged**
